multiqc/modules/pairtools/pairtools.py

```diff
@@ -0,0 +1,522 @@
+            'ylab': '# Reads',
+            'cpswitch_counts_label': 'Number of Reads',
+            'data_labels': data_labels
+        }
+
+        # annotate read orientations with nice colors:
+        keys_annotated = OrderedDict()
+        for key in orientation_keys:
+            name = self.params["pairs_orientation_names"][key]
+            color = self.params["pairs_orientation_colors"][key]
+            keys_annotated[key] = {'color': color, 'name': name}
+
+        return bargraph.plot(
+                [_data[_d] for _d in dist_ranges],
+                [keys_annotated for _d in dist_ranges],
+                pconfig=config,
+            )
+
+
+
+    def pairs_with_genomic_separation(self):
+        """
+        number of cis-pairs with genomic separation
+        # dist_freq/56234133-100000000/-+
+        """
+
+        dist_freq_field = "dist_freq"
+        data_cats = ["avg", "FF", "FR", "RR", "RF"]
+        # extract pair orientations types - as they'll be used here many times
+        porient_names = self.params["pairs_orientation_names"]
+
+        # Initialize a data structure for the plot
+        _data = {cat: {} for cat in data_cats}
+
+        for s_name in self.pairtools_stats:
+            for cat in data_cats:
+                _data[cat][s_name] = {}
+            # pre-calculate geom-mean of dist-bins for P(s):
+            _dist_bins = np.asarray(self.pairtools_stats[s_name]['dist_bins'])
+            # extract contacts by distance ranges and orientation types
+            sample_dist_freq = self.pairtools_stats[s_name][dist_freq_field]
+
+            # contacts by orientation and distance, and their average summary,
+            _summary = {}
+            _summary["avg"] = np.sum([sample_dist_freq[po] for po in porient_names],axis=0).astype(float)
+            for po, po_name in porient_names.items():
+                _summary[po_name] = np.asarray(sample_dist_freq[po]).astype(float)
+
+            # this is wrong - should be chromsizes based :
+            if self.pairtools_stats[s_name]["chromsizes"]:
+                # normalize contacts by distance with the theoretical # of pairs in a range
+                _areas = contact_areas( _dist_bins, scaffold_length=2_000_000_000_000 )
+                for cat in data_cats:
+                    _summary[cat] = _summary[cat]/_areas
+
+            # assign geometric mean distance to every distance interval
+            _dist_bins_geom = np.sqrt(_dist_bins[1:]*_dist_bins[:-1])
+            # fill in the data for XY-line plotting
+            # i.e. dict by samples of dicts by dist (X), of (normalized) counts (Y):
+            for cat in data_cats:
+                _data[cat][s_name] = dict(zip(_dist_bins_geom[1:], _summary[cat][2:]))
+
+                log.critical(_dist_bins_geom)
+
+        pconfig = {
+            'id': 'broom_plot',
+            'title': 'Pairs by distance and by read orientation',
+            'xlab': 'Genomic separation (bp)',
```

Comment:
Need a y-axis label to pass MultiQC `--lint`:

```suggestion
            'xlab': 'Genomic separation (bp)',
            'ylab': 'Number of reads',
```